editor/modes/visual-editor/block.js

```diff
@@ -11,84 +11,116 @@ import Toolbar from 'components/toolbar';
+			onChange( {
+				attributes: {
+					...block.attributes,
+					...attributes
+				}
+			} );
 		}
-	}
 
-	// Disable reason: Each block can receive focus but must be able to contain
-	// block children. Tab keyboard navigation enabled by tabIndex assignment.
+		function maybeDeselect( event ) {
+			// Annoyingly React does not support focusOut and we're forced to check
+			// related target to ensure it's not a child when blur fires.
+			if ( ! event.currentTarget.contains( event.relatedTarget ) ) {
+				onDeselect();
+			}
+		}
 
-	/* eslint-disable jsx-a11y/no-static-element-interactions */
-	return (
-		<div
-			tabIndex="0"
-			onFocus={ onSelect }
-			onBlur={ maybeDeselect }
-			onKeyDown={ onDeselect }
-			onMouseEnter={ onMouseEnter }
-			onMouseLeave={ onMouseLeave }
-			className={ className }
-		>
-			{ ( isSelected || isHovered ) && <BlockMover uid={ block.uid } /> }
-			<div className="editor-visual-editor__block-controls">
-				{ isSelected && <BlockSwitcher uid={ block.uid } /> }
-				{ isSelected && settings.controls ? (
-					<Toolbar
-						controls={ settings.controls.map( ( control ) => ( {
-							...control,
-							onClick: () => control.onClick( block.attributes, setAttributes ),
-							isActive: () => control.isActive( block.attributes )
-						} ) ) } />
-				) : null }
+		// Disable reason: Each block can receive focus but must be able to contain
+		// block children. Tab keyboard navigation enabled by tabIndex assignment.
+
+		/* eslint-disable jsx-a11y/no-static-element-interactions */
+		return (
+			<div
+				ref={ this.bindBlockNode }
+				tabIndex="0"
+				onFocus={ onSelect }
+				onBlur={ maybeDeselect }
+				onKeyDown={ onDeselect }
+				onMouseEnter={ onMouseEnter }
+				onMouseLeave={ onMouseLeave }
+				className={ className }
+			>
+				{ ( isSelected || isHovered ) && <BlockMover uid={ block.uid } /> }
+				<div className="editor-visual-editor__block-controls">
+					{ isSelected && <BlockSwitcher uid={ block.uid } /> }
+					{ isSelected && settings.controls ? (
+						<Toolbar
+							controls={ settings.controls.map( ( control ) => ( {
+								...control,
+								onClick: () => control.onClick( block.attributes, setAttributes ),
+								isActive: () => control.isActive( block.attributes )
+							} ) ) } />
+					) : null }
+				</div>
+				<BlockEdit
+					isSelected={ isSelected }
+					attributes={ block.attributes }
+					setAttributes={ setAttributes }
+					insertBlockAfter={ onInsertAfter }
+				/>
 			</div>
-			<BlockEdit
-				isSelected={ isSelected }
-				attributes={ block.attributes }
-				setAttributes={ setAttributes }
-				insertBlockAfter={ onInsertAfter }
-			/>
-		</div>
-	);
-	/* eslint-enable jsx-a11y/no-static-element-interactions */
+		);
+		/* eslint-enable jsx-a11y/no-static-element-interactions */
+	}
 }
 
 export default connect(
 	( state, ownProps ) => ( {
+		order: state.blocks.order.findIndex( uid => ownProps.uid === uid ),
```

Comment:
We haven't enabled the [polyfill](http://babeljs.io/docs/usage/polyfill/), so this will error in IE11. We might just consider enabling the polyfill, but in this case couldn't this just be:

```js
order: state.blocks.order.indexOf( ownProps.uid )
```